pwndbg/gdblib/disasm/arch.py

```diff
@@ -668,7 +669,9 @@ def _resolve_target(self, instruction: PwndbgInstruction, emu: Emulator | None,
                         addr = resolved_addr
                     else:
                         page = pwndbg.gdblib.vmmap.find(resolved_addr)
-                        if page and page.execute:
+                        # When debugging a remote QEMU target, the page permissions are not accurate.
+                        # In this case, if the candidate address is mapped at all, just go with it.
+                        if page and (pwndbg.gdblib.remote.is_remote() or page.execute):
```

Comment:
Which check is 'heavier'? Shall we check for `page.execute` first and only then call `is_remote()`? :)
```suggestion
                        if page and (page.execute or pwndbg.gdblib.remote.is_remote()):
```

Reply:
True, the check for is_remote may make a call to a GDB command (or at least does a dictionary lookup on @cache decorator- I flipped the order